src/NServiceBus.RabbitMQ/Sending/MessageDispatcher.cs

```diff
@@ -21,14 +20,17 @@ public Task Dispatch(TransportOperations operations, ContextBag context)
 
             try
             {
-                var tasks = new List<Task>(operations.UnicastTransportOperations.Count() + operations.MulticastTransportOperations.Count());
+                var unicastTransportOperations = (List<UnicastTransportOperation>)operations.UnicastTransportOperations;
+                var multicastTransportOperations = (List<MulticastTransportOperation>)operations.MulticastTransportOperations;
```

Comment:
I don't think we should make this cast. Although we know that *currently* core is passing in a `List<T>`, that might change, and the compiler will never tell us we have a problem. I'd be happy with:

```C#
var unicastTransportOperations = operations.UnicastTransportOperations as List<UnicastTransportOperation> ??
  operations.UnicastTransportOperations.ToList();
```